### PR TITLE
Leo Add ability to create a team from the “Add Team” option of User Management

### PR DESCRIPTION
**Pull request by Chuehleo**
# Description
Screenshot 2023-07-12 001931
…
## Related PRS (if any):
This frontend PR is related to [#453](https://github.com/OneCommunityGlobal/HGNRest/pull/453) backend PR
…

## Main changes explained:
- Update AddTeamPopup.jsx for including new "create new team" button
- Able to create a new team in the add team popup
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. Admin/Owner Login → Other Links → User Management → Create New User → Assign Team 
5. Check if the create new team button is able to create a new team and add team immediately

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/122568562/78fb1346-b78d-406d-b387-56e77b42e5ec

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Review by beblicarl — DISMISSED** [review]

**Review by tdkent — DISMISSED** [review]
Local testing of the PR branch confirms 

- a **Create New Team** input has been added to the **Add Team** popup form in User Management > Create New User.
- using the input successfully creates a new team, which is added to the Teams list.
- the new user can subsequently be added to the team in the same **Add Team** form.

Issues encountered which may be worth addressing:

- the **Create New Team** functionality does not appear to have been added to the Permissions Management page, though as noted in the description this may require a separate PR to implement.
- The **Add Team** form styles might be updated to make the "add team" and "create team" functions easier to differentiate (ie divide the form into separate sections with "Add To Team" and "Create New Team" headers).
- Adding a new team is confirmed via an alert that must be manually cleared by the user. This might be improved with a toast alert or text alert that appears in the form itself.
- The **Add Team** form styling becomes strange between 768-992px screen widths (see below).


997-add-team

Testing Video:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/03d6b2e8-fcd2-4bce-9557-735e7999bfc8

**Comment by AnishPandita** [comment]
Hi @Chuehleo I tested the PR and although the main functionality works well. There was a issue regarding duplicate name entries. Also I agree with @tdkent suggestion about making the add team success popup a toast or alert. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111321945/72215468-db5a-487d-8915-a5d80639491c

Here you can add same name again creating duplicate entries for same teams.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111321945/86259f56-d28d-4be9-ba22-90c4971d9cb4

**Review by BaileyM96 — CHANGES_REQUESTED** [review]
Hello @Chuehleo I tested your Add Team Modal and everything works great!

## Issues

The only issue/question I have is that when on "Add Team" modal the confirm input field is kind of confusing I was not sure what I was supposed to do with that for a few seconds. Maybe we can make it where the confirm button is just located next to the cancel button.

## Screenshots
After creating the team for the user
PR #997 Add Team to creating user

The UI issue
PR #997 Issue

**Comment by iTvX** [comment]
Work as described.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/a68e6667-225c-4948-974d-97a3f6c9b7e0

 duplicate name entries have some issue like others said. I also have one more suggestion, When I first entered this interface, my initial instinct was to type a team name into the first line. However, it prompted me with an error message that read, "Hey, You need to pick a team first!". I believe this error message isn't suitable. In my opinion, the correct error message should be, "This team isn't found, you need to create a team first."

**Review by iTvX — CHANGES_REQUESTED** [review]
Work as described.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/24f45cde-0600-4b49-80c0-cf17d8457349



However, duplicate name entries have some issue like others said.

I also have one more suggestion, When I first entered this interface, my initial instinct was to type a team name into the first line. However, it prompted me with an error message that read, "Hey, You need to pick a team first!". I believe this error message isn't suitable. In my opinion, the correct error message should be, "This team isn't found, you need to create a team first."

**Review by scoopstroop92 — COMMENTED** [review]
Hello @Chuehleo : The functionality works as expected. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/5d9966d8-21f8-4911-8c25-f5b61e095ea6

**Bug** : 
Duplicate team creation is allowed.  It should ideally throw "team name already exist" error


**UI Suggestions :** 
User clicks on the Assign Team , the Assign Team pop up shows up
Existing UI
image

Proposed UI : 
image

**1. User clicks on use existing team :** 
image

**2. User clicks on create new team :** 
image
Now user comes back to use existing team page 
image

**Comment by freecode23** [comment]
Agree with everyone on the UI issue. But otherwise the code works fine.
I tried logging in as user with the team assigned and it works.

For the UI, I'm thinking of this:
Screenshot 2023-07-13 at 9 44 35 PM

but this would mean that we need to add the logic in the backend if team does not exist yet, then create it before assigning

**Review by luisarevalo21 — CHANGES_REQUESTED** [review]
Hello @Chuehleo,

I tested your PR and works as intended great job! 

As others have pointed out theres a few UI change that can make the app much better.
A few I suggest are: 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/62eac0f0-32cb-499f-8ee8-950ffa75e471

- First the searching for a team should have text inside with "search for team" instead of being blank. Could be a separate PR 
- Next if the search team had a text an a new team is create, the text inside of the search team input should be set to "" have placeholder text of "search for team" (text can be something else just used this as an example).' See video for example

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/b2b611f2-9871-4589-9256-b2070d2e7742

- When a team is created a pop up should appear saying "team was successfully created" 
- When confirming the user to a team the modal should should disappear or a popup message saying "member added to team dismiss modal to continue." (text can be something else just used this as an example).

File: src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx [hunk 1]
```diff
@@ -41,14 +73,20 @@ const AddTeamPopup = React.memo(props => {
             Confirm
           </Button>
         </div>
-        <div>
-          {!isValidTeam && selectedTeam && (
```
Comment (AaronPersaud) [inline src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx:45]: Why was this block removed? (45-47) I feel like this message is necessary if the user chooses a team that the person is already assigned to.

File: src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx [hunk 2]
```diff
@@ -18,18 +31,37 @@ const AddTeamPopup = React.memo(props => {
       onValidation(false);
     }
   };
+
   const selectTeam = team => {
     onSelectTeam(team);
     onValidation(true);
   };
 
+  const onCreateTeam = async () => {
```
Comment (AaronPersaud) [inline src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx:40]: You should probably add another case in this function to check if there already exists a team with the given name.

File: src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
 
 const AddTeamPopup = React.memo(props => {
+  const dispatch = useDispatch();
   const closePopup = () => {
     props.onClose();
   };
 
   const [selectedTeam, onSelectTeam] = useState(undefined);
   const [isValidTeam, onValidation] = useState(true);
+  const [isValidNewTeam, onNewTeamValidation] = useState(true);
```
Comment (AaronPersaud) [inline src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx:15]: What is the point of this variable? Every time it is called its value is just getting modified, but is not used anywhere.

File: src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
+
+  // states and onrs for the new team form
+  const [newTeamName, setNewTeamName] = useState('');
+  const [newTeamIsActive, setNewTeamIsActive] = useState(true);
+
+  const onNameChange = (e) => {
+    setNewTeamName(e.target.value);
+    onNewTeamValidation(true);
```
Comment (AaronPersaud) [inline src/components/UserProfile/TeamsAndProjects/AddTeamPopup.jsx:23]: (Line 23) This setter gets called every time the new team name changes i.e each time a letter is typed. This seems costly and redundant, and unnecessary as its not being used for anything. You can probably remove it.

**Review by AaronPersaud — CHANGES_REQUESTED** [review]
Hi, I've tested and it looks good!

<b>Creating new team from "Create New User" Modal</b>

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/10265513/2bb5e5ef-72f9-4998-a6c4-fba2829e04cb

- can create new team
- can add user to existing team
- error thrown when trying to add to team that doesn't exist

However, I have a few suggestions.

1.) Users can create a team with a name that's already in use. This seems like a bad idea as the only way to differentiate between same named teams is their ID but I don't believe the user has access to that. Add an error popup that is displayed when a user clicks "Create New Team" with an already existing team name. See my comment for line 40.

2.) When a user tries to add a team that the user is already part of, the error "Hey, You need to pick a team first!" pops up which doesn't make sense for this scenario. You can probably fix this by adding lines 45-47 back and possibly changing the condition logic.

3.) Generally, dropdowns have a functionality that allow you to navigate through the options using the up and down arrow keys. You would need to add `onKeyUp` and `onKeyDown` attributes to the input component that change the selected value.

The last suggestion is just a personal suggestion, and it seems that the dropdown was already there before this ticket so it probably can be done in a separate PR.

Another suggestion I thought of that fixes problem 2 is disabling the "Confirm" button until the team is a valid team name (exists & not already added). This would not require the toasts for the 2 scenarios that occur when clicking confirm.

**Review by siddharthmohite — DISMISSED** [review]
The functionality works as expected. But as others have pointed out it is confusing for a first-time user to understand it. The UI Suggestion given by @ramyaram2092 is amazing!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/37925823/d61fb58f-d76d-4c3b-8455-971de04e7b88

**Comment by Jiangwei-shi** [comment]
I tested your PR and works as intended great job!
image
image

**Review by HectorAgudelo — DISMISSED** [review]
The app was tested following the instructions on the PR, and it performs as expected. it works while creating an user, and also when the user is already created.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/37153730/26faeef0-a21b-4f3a-92ab-1156a48d721c

**Review by shiva2096 — DISMISSED** [review]
The app was tested following the instructions on the PR and It performs as expected. 
UI suggestion above from the team looks better, but this also works fine.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26687177/1957e72c-7ee9-457b-ac9a-9e674d95a901

**Comment by one-community** [comment]
> I also have one more suggestion, When I first entered this interface, my initial instinct was to type a team name into the first line. However, it prompted me with an error message that read, "Hey, You need to pick a team first!". I believe this error message isn't suitable. In my opinion, the correct error message should be, "This team isn't found, you need to create a team first."

Yes, this message needs to be updated. 

1. "Hey, You need to pick a team first!" should show if they click "Confirm" without typing anything. 
2. If they type something that does not exist, please have it say, "Oops, this team doesn't exist! Create it if you want it."

**Review by mahamakhan — DISMISSED** [review]
The mentioned function works fine with new and existing users from the admin user.

**Comment by igorrochadasilva** [comment]
Hi @Chuehleo !
I hope you are well =D

I tested your PR and works as intended great job!

997_01_create-team
997_02_team-created
997_03_team-added

**Review by sxiong5 — DISMISSED** [review]
Hey, I tested your changes, and they worked as intended.

Screenshot 2023-07-20 at 14 26 36
Screenshot 2023-07-20 at 14 26 45
Screenshot 2023-07-20 at 14 26 51

# Suggestion
(Just a suggestion to improve the UI)
The operation, for now, is a little bit complicated, and the UI might be confusing.
The following is a component demo from MUI. We could learn from it since we have the same requirements.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/598d2e4d-e041-4de0-a60e-7b8194d36e46

**Comment by DavidC0126** [comment]
Hi @Chuehleo I tested your PR and everything works as intended. I am able to create a new team or assign to an existing team immediately. I also agree with the UI suggestions in this post. Nice work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/c7101b12-b865-4f8c-b022-4d4654149e67

pr453

**Comment by ramyaram2092** [comment]
Hello @Chuehleo  could you check my comment in the backend PR  453?

**Comment by Shreyj14** [comment]
Hi @Chuehleo 

I tested this PR which is a dependency for #453 in the backend, it works as intended, and I was able to create the team. I can see that certain changes have been implemented which were part of the suggestions from the comments above. The changes I noticed which look good are -
1) When confirm is clicked without picking a team it displays the message "Hey, you need to pick a team first"
2) If a team exists it throws a message alerting that the team already exists.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118795427/b6a3a250-6171-40b9-8a37-0aab0b680d72

**Review by scoopstroop92 — DISMISSED** [review]
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/e9e3a4bd-f3ad-4944-964c-c3b19a274ab8

Works well. All the suggestions and feed back has been addressed 👍 

Testing Video

**Review by vemanamadhu — CHANGES_REQUESTED** [review]
Hello Leo,

Good job on creating the UI and the flow for creating a team while creating a user.
I have a couple of suggestions. 
I created a new team for the user but then I got confused for a second on what to do. 

I had a couple of suggestions about the flow of things
1. I was very confused about what to do after creating a team. I then realized that we had to go and add the team that we created back to the add-to team prompt. This flow in my opinion is not optimal for user experience. I feel like we need to revisit this flow.
Suggestions on different flows
- Once we create a team it auto-assigns the newly created team to the user
- We can remove the create new team prompt and incorporate that into the add-to-team prompt. If the team name we are typing doesn't exist we can show the create new team link in the drop down itself. We can do something like shown in the screenshot below.
- 
PR # 997 review_suggestions

2. I feel like the new team was created successfully and the team already exists message should show up in the pop-up and not on the side of the screen. It can be missed if you have a big screen you are viewing the app on.

Please let me know what you think. Apart from that everything seems to be working as expected. Good job

**Comment by one-community** [comment]
> * Once we create a team it auto-assigns the newly created team to the user
> * We can remove the create new team prompt and incorporate that into the add-to-team prompt. If the team name we are typing doesn't exist we can show the create new team link in the drop down itself. We can do something like shown in the screenshot below.
> 
> PR # 997 review_suggestions
> 
> 2. I feel like the new team was created successfully and the team already exists message should show up in the pop-up and not on the side of the screen. It can be missed if you have a big screen you are viewing the app on.

These are great suggestions, please integrate them

**Comment by Chuehleo** [comment]
> These are great suggestions, please integrate them

No problem. I'll start working on integrating these suggestions.

**Comment by Chuehleo** [comment]
> Hello Leo,
> 
> Good job on creating the UI and the flow for creating a team while creating a user. I have a couple of suggestions. I created a new team for the user but then I got confused for a second on what to do.
> 
> I had a couple of suggestions about the flow of things
> 
> 1. I was very confused about what to do after creating a team. I then realized that we had to go and add the team that we created back to the add-to team prompt. This flow in my opinion is not optimal for user experience. I feel like we need to revisit this flow.
>    Suggestions on different flows
> 
> * Once we create a team it auto-assigns the newly created team to the user
> * We can remove the create new team prompt and incorporate that into the add-to-team prompt. If the team name we are typing doesn't exist we can show the create new team link in the drop down itself. We can do something like shown in the screenshot below.
> 
> PR # 997 review_suggestions
> 
> 2. I feel like the new team was created successfully and the team already exists message should show up in the pop-up and not on the side of the screen. It can be missed if you have a big screen you are viewing the app on.
> 
> Please let me know what you think. Apart from that everything seems to be working as expected. Good job

For the suggestion create new team prompt and incorporate that into the add-to-team prompt, do you mean that the drop down list including the create new team link should be able to pop the create new team modal up when the user click it?

**Comment by vemanamadhu** [comment]
> So what I was thinking was:
There can be one prompt called Add Team.
When we search for a team --> A "create new team ->" option shows up in the drop-down. It can become active if a team with the same name doesn't exist.

So once we click on the "create new team ->" in the dropdown -- the backend process to create a new team can be triggered. Once the team gets created --> We can get a notification saying "Team created successfully".

And then there can be a confirm and cancel button at the bottom which confirms that we have assigned a team.

**Review by eduardocodes — DISMISSED** [review]
Hi @Chuehleo 

I tested this PR and everything works as intended. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86692035/21b56847-ebb2-425d-8c5f-55b5abb1fa98

**Review by AaronPersaud — DISMISSED** [review]
Hi, I've tested it again and you've addressed all my requests! Everything looks good!

1.) You cannot create duplicate team names anymore

<b>Creating team with a name that exists</b>

image

2.) No incorrect error when trying to add to a team that user is already a part of 

<b>Adding to team new user is already on</b>

image

- no message appears at all if you attempt this, but this is a very specific edge case where the toast isn't super necessary

3.) Outside of scope of the PR (`onKeyUp` and `onKeyDown` for dropdown)

Great job!

**Review by Indujaks — DISMISSED** [review]
Tested Changes, Works as Expected!
Screenshot 2023-07-27 at 4 57 12 PM

**Review by yaow62023 — DISMISSED** [review]
After testing this PR, I found that the functionalities mentioned in the description are working as expected. However, I have a couple of suggestions:
1. After creating a team, is it possible to automatically join the team? This would eliminate the need for a second step to assign oneself to the team.
2. Shouldn't team names be unique? In the video, I was able to create multiple teams with the same name and join them. This could potentially lead to conflicts in the future.

Feature test video:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/146370da-1978-4d75-a6f4-5bbaa0b466e3

Video for add to team which same name:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/838942ac-10a4-4f0e-b49b-5d82c9f1f923

**Review by zuhang2000 — DISMISSED** [review]
I have tested both the front-end and back-end. The functionality are working as expected.

More specifically, I did the test on "Create New Team" and "Add to a Team". I created several new teams with different format of team name and try to see whether it can successfully detect all kinds of duplicate team name. It works as expected and it would disallow creating a team with the same name as an existing team.

Also as @one-community suggested previously, I think those suggestions would make this feature even better. 

But I do notice many people suggest about adding the user to the team once the team is created. I think it's gonna be a problem if the admin account is creating a new user and at the same time also creating a new team. Which person should be added to this new team? The new user or the admin or both of them? I think more considerations should be made on how to handle this case. But if creating a new team elsewhere, adding the user to the new team once the team is created would be a great feature.

Screen Shot 2023-07-28 at 5 17 49 PM
Screen Shot 2023-07-28 at 5 24 55 PM

In this picture, I'm trying to add the person who is already in the team to the team again. It is great that the error pops up which does not allow me to do so. It's just gonna be better if the error message could be "The person is already in the team". I think it would be confusing in the case that if there are a lot of people in the team, it would be hard to figure out whether I made the name wrong or the person is already in the team. This is only my suggestion. All the functionality is working as expected. 
Screen Shot 2023-07-28 at 5 28 29 PM

Good job!

**Review by mahamakhan — DISMISSED** [review]
Tested the functionality and was able to create a team and add it successfully. 

image
Screenshot 2023-07-29 200615
Screenshot 2023-07-29 200519
Screenshot 2023-07-29 200714

**Review by ThapeloMasasa — DISMISSED** [review]
The functionality works as intended and the changes made to the code are great

Screenshot 2023-07-31 at 9 45 44 PM

Screenshot 2023-07-31 at 9 46 04 PM

**Review by winghojackyli — DISMISSED** [review]
Hello @Chuehleo , I can create team and add the new user into the new created team while creating user. It works well with the backend https://github.com/OneCommunityGlobal/HGNRest/pull/453


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/fb2f9b4a-5004-4db5-ac30-99c295a620d5

**Comment by veronicacheng2** [comment]
Hi @Chuehleo, I tried testing the PR but 'Create New Team' input and button do not show up. 

image

**Comment by Chuehleo** [comment]
> Hi @Chuehleo, I tried testing the PR but 'Create New Team' input and button do not show up.
> 
> image

Hi @veronicacheng2 , I have change the code to what @vemanamadhu had requested, so there won't be a create new team button, just simply type the team you want to create and the dropdown list will allow you to create it if it doesn't exist. The video is also provided in the description of this PR.

**Review by veronicacheng2 — APPROVED** [review]
> > Hi @Chuehleo, I tried testing the PR but 'Create New Team' input and button do not show up.
> > image
> 
> Hi @veronicacheng2 , I have change the code to what @vemanamadhu had requested, so there won't be a create new team button, just simply type the team you want to create and the dropdown list will allow you to create it if it doesn't exist. The video is also provided in the description of this PR.

Hi @Chuehleo, thank you for your prompt reply. I have tested again according to the new instructions. Everything works perfectly.

 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95119686/b6f2d9bf-858d-48ff-841b-8f145c4ef48b

**Review by harshidad — DISMISSED** [review]
This code working as expected. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/f03921fa-5c85-4053-acbe-341c23fe70e9

**Review by winghojackyli — APPROVED** [review]
Hi @Chuehleo , I've tested this updated PR and it works as intended. The new create team method works well with the UI. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/73fef416-7f71-4e04-9c7f-8e2bd65d03f5

**Comment by Jiyuanxia** [comment]
Hi @Chuehleo, I have reviewed your pull request and tested it. It works as you intended. Well done!

Screenshot 2023-08-24 211604
Screenshot 2023-08-24 211617

**Review by yunshan111 — DISMISSED** [review]
Tested this PR, LGTM!

**Comment by Shubhu0** [comment]
Reviewed the PR and all its changes, works as intended

**Comment by Shubhu0** [comment]
Reviewed this PR. Working as intended. Great job!!

**Review by zuhang2000 — DISMISSED** [review]
Hi Leo! I'm following up on my previous review on your PR. This PR now works as expected! This functionality and new feature is good. Here below is my recording on the test.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/7147f0d9-dd27-486c-bf1c-0a71ad9f87b2

**Comment by one-community** [comment]
Be sure to pull the latest code so these formatting issues that have already been fixed are included: 
image

If there were a way to add that "Create" button elegantly to the right of the word "Teams", that would be nice too. If keeping it below, please change label to "Create Team" and make it the same size as the "Assign Team" button

**Comment by Chuehleo** [comment]
> Be sure to pull the latest code so these formatting issues that have already been fixed are included: image
> 
> If there were a way to add that "Create" button elegantly to the right of the word "Teams", that would be nice too. If keeping it below, please change label to "Create Team" and make it the same size as the "Assign Team" button
Do you mean the "Create" button of the create user modal or the one after I click the "Assign Team" button. For the lastest code, there is only a dropdown option to create the team so there won't be be a "Create" button.
Screenshot 2023-09-07 121127
unityGlobal/HighestGoodNetworkApp/assets/122568562/ac42011d-0c09-4a34-ba26-eceb17765920)

**Review by Shiwani99 — DISMISSED** [review]
Hi,
Great work. I tested your PR and it works as expected. I am attaching a screen recording of how it looks in my end.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/6732eaa5-0a80-4088-9c5e-6960fa273c23

**Review by wantingxu7 — DISMISSED** [review]
Hi,
I've tested and it works as intended. Please find in the recording below.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/2b2764cb-9372-4973-8e17-934c9d06f7e5

**Comment by Shubhu0** [comment]
Hi @Chuehleo, I have reviewed your pull request and tested it and it works as you intended. Well done!

image

**Comment by RheaWu1212** [comment]
'create new team' not displayed 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109626741/2d064adc-2b89-4746-acf8-7ceaa3e258e0

**Review by JeffLi117 — DISMISSED** [review]
Changes work as intended. 

When the text within the search bar does not directly match a team name, there is an option to "Create new team" with said name (testing997 team was created already, I took this screenshot after I had tested everything else): 
image

Successfully created a team (with respective toast):
image

Successfully assigned the newly created team to my new user:
image

Double checking that my new team is in the system and the new user is also listed within said team:
image

Great job!

**Review by yi-lin-1234 — DISMISSED** [review]
Hi, I tested this PR with #453, and it works as intended. Great work!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92185695/f1bab9a8-3fa3-401e-8731-924d415e3330

**Review by StrawberryCalpico — DISMISSED** [review]
Hi I tested this PR, it worked perfectly, Good job

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114443664/ab49a025-ad43-4c27-9704-b2849dc40f02

**Review by malikjahanzaib — APPROVED** [review]
Hi @Chuehleo, I tested this PR locally and it works great. I was able to create a new team from the user creation popup and verified the team existed in the "Teams" tab. Great job!!!

Test Rec:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46221334/69814dbd-f39f-42b5-b867-76b88f0cfa41

**Comment by kavya151** [comment]
Hi @Chuehleo I have tested this PR locally and it works great. I was able to create new team and also the created team exists in the team and doesn't allow creating duplicate teams.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61841929/35066222-9d31-43e8-b879-29f1a06ea18d

**Comment by kavya151** [comment]
Hi @Chuehleo . I have tested this PR locally and it allows me to successfully create  new team and also the created team is visible in the teams.

Uploading Screen Recording 2023-10-25 at 5.12.02 PM.mov…

**Comment by emunkh404** [comment]
Hi, I tested your work, Great job!
image

**Comment by kavya151** [comment]
Thank you

**Review by navneeeth — CHANGES_REQUESTED** [review]
Hey @Chuehleo! The functionality works as expected and the team creation happens hassle-free and the user can be added to the newly created team as verified from the Teams page. Good implementation!

## Test Video

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/70ab7fe6-805b-4084-8983-160cb39689bc


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/64e394b5-c36f-411d-99bd-b86dee4ad336

## Issue
However, right after the team is created, there is an issue with clicking on 'Confirm' which does not recognize the recently created team until it is searched again as shown here: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/8d4d7748-8a6b-4dc1-a187-d475399023ea

Please fix this such that after team creation, the 'Confirm' functionality works without having to search again for the team name. Thank you!

**Review by navneeeth — APPROVED** [review]
Hello, @Chuehleo! Great job on addressing the issue! The functionality works as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/d45c6afd-d48a-42f6-b01e-8fc036f98691

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**